### PR TITLE
feat: add changePassword (parent app) endpoint

### PR DESCRIPTION
**Pull request by IsaacHatton**
Add the endpoint required to change the account password for the parent app

### CODE REVIEW

File: src/core/parentClient.ts [hunk 1]
```diff
@@ -88,4 +88,29 @@ export class ParentClient extends BaseClient {
...
+   * @param current Current password
+   * @param new New password
+   * @returns whether it was successful or not
+   */
+    async changePassword(this: ParentClient, current: string, newPassword: string): Promise<ChangePasswordResponse>{
+      const formData = new URLSearchParams();
+      formData.append("current", current);
+      formData.append("new", newPassword);
+      formData.append("repeat", newPassword);
+      const headers = new Headers({
```
Comment (jamesatjaminit) [inline src/core/parentClient.ts]:
```suggestion
```
I'd prefer this simply as a JSON object passed straight to the headers parameter, to keep it consistent with the rest of the lib.

File: src/core/parentClient.ts [hunk 2]
```diff
@@ -88,4 +88,29 @@ export class ParentClient extends BaseClient {
     }
     throw new Error("No pupil with specified ID returned");
   }
+    /**
+   * Changes the login password for this parent's account
+   * @param current Current password
+   * @param new New password
+   * @returns whether it was successful or not
```
Comment (jamesatjaminit) [inline src/core/parentClient.ts]:
```suggestion
   * @returns Whether the request was successful
```

File: src/core/parentClient.ts [hunk 3]
```diff
@@ -88,4 +88,29 @@ export class ParentClient extends BaseClient {
     }
     throw new Error("No pupil with specified ID returned");
   }
+    /**
+   * Changes the login password for this parent's account
```
Comment (jamesatjaminit) [inline src/core/parentClient.ts]:
```suggestion
   * Changes the login password for the current parent account
```

File: src/core/parentClient.ts [hunk 4]
```diff
@@ -88,4 +88,29 @@ export class ParentClient extends BaseClient {
...
+   * @param current Current password
+   * @param new New password
+   * @returns whether it was successful or not
+   */
+    async changePassword(this: ParentClient, current: string, newPassword: string): Promise<ChangePasswordResponse>{
```
Comment (jamesatjaminit) [inline src/core/parentClient.ts]:
```suggestion
    async changePassword(currentPassword: string, newPassword: string): Promise<ChangePasswordResponse> {
```
Unsure why `this` is required. And current should be renamed to `currentPassword` for consistency

File: src/types.ts [hunk 5]
```diff
@@ -490,6 +490,8 @@ export interface PupilFieldsData {
 
 export type PupilFieldsResponse = ClassChartsResponse<PupilFieldsData, []>;
 
+export type ChangePasswordResponse = ClassChartsResponse<[], []>;
```
Comment (jamesatjaminit) [inline src/types.ts:493]: Does the API not give any feedback to whether the request was successful? And any meta value?

File: src/core/parentClient.ts [hunk 6]
```diff
@@ -88,4 +88,29 @@ export class ParentClient extends BaseClient {
     }
     throw new Error("No pupil with specified ID returned");
   }
+    /**
+   * Changes the login password for this parent's account
+   * @param current Current password
+   * @param new New password
```
Comment (jamesatjaminit) [inline src/core/parentClient.ts]:
```suggestion
   * @param newPassword New password
```

File: src/core/parentClient.ts [hunk 7]
```diff
@@ -88,4 +88,29 @@ export class ParentClient extends BaseClient {
     }
     throw new Error("No pupil with specified ID returned");
   }
+    /**
+   * Changes the login password for this parent's account
+   * @param current Current password
```
Comment (jamesatjaminit) [inline src/core/parentClient.ts]:
```suggestion
   * @param currentPassword Current password
```

**Review by jamesatjaminit — CHANGES_REQUESTED** [review]

File: src/types.ts (hunk 5 above)
Comment (IsaacHatton) [reply src/types.ts:493]: The api returns this { success: 1, data: [], meta: [] }

Comment (jamesatjaminit) [reply src/types.ts:493]: Ah fair enough

**Comment by IsaacHatton** [comment]
Is there something I need to press to approve changes here?

**Comment by jamesatjaminit** [comment]
The piece of code implementing headers is now invalid, you haven't moved it to the headers parameter and the object is still [half there](https://github.com/classchartsapi/classcharts-api-js/pull/40/files#diff-25acbb849e48788cf8c74dc5a3b28eaf9fc1c0f15c149c858c768c2b7457718bR102). Please make sure you check for errors and test the library (`deno test -A`).

Preferably run `deno fmt` as well, but I can do that if needed.

And no, you don't need to approve changes, that's a button I press :)

**Comment by IsaacHatton** [comment]
Apologies for managing to leave an invalid header element and incorrect formatting.

I have ran deno test -A and deno fmt and it seems to all be fine, and used it successfully to change the password of the parent account.

Are there any further changes I need to make?

**Review by jamesatjaminit — APPROVED** [review]

**Comment by jamesatjaminit** [comment]
Thanks :)

**Pull request merged**